metricflow/dataflow/optimizer/predicate_pushdown_optimizer.py

```diff
@@ -458,6 +459,11 @@ def visit_join_conversion_events_node(self, node: JoinConversionEventsNode) -> O
             )
         )
 
+    def visit_join_to_custom_granularity_node(  # noqa: D102
```

Comment:
@tlento I might need some help to figure this one out correctly - I looked at the logic in `visit_join_to_time_spine_node()` to try to emulate that, but I didn't totally understand it, so it would be helpful for me if we could talk through it!

Reply:
Yeah, this is complicated. As discussed offline, raising here and disabling the optimizer seems like our best option.